src/dialog/src/Dialog.js

```diff
@@ -50,28 +50,88 @@ const animationStyles = {
 
 class Dialog extends React.Component {
   static propTypes = {
+    /**
+     * Composes the Overlay component as the base.
+     */
     ...Overlay.propTypes,
+
+    /**
+     * Children can be a node or a function accepting `({ close })`.
+     * See an example to understand how this works.
+     */
+    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+
+    /**
+     * Title of the Dialog. Titles should use Title Case.
+     */
     title: PropTypes.node,
+
+    /**
+     * Props passed through to the primary button.
+     * Passing this object will show the button and cancel button.
+     * You should pass `children` and `onClick`.
+     */
+    primaryButton: PropTypes.object,
+
+    /**
+     * Label of the cancel button, shown when primaryButton is passed.
+     * You should not have to change this in most cases.
+     */
+    cancelLabel: PropTypes.node,
+
+    /**
+     * Only effective when when passing primaryButton.
+     */
+    hideCancelButton: PropTypes.bool,
+
+    /**
+     * Width of the Dialog.
+     */
     width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    hasCloseIcon: PropTypes.bool,
+
+    /**
+     * The space above the Dialog.
+     * This offset is also used at the bottom when there is not enough space
+     * available on screen — and the Dialog scrolls internally.
+     */
+    topOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
This can't be a number because I don't think unitless values will work in the CSS calc below.

Reply:
Ah, we need to do add px then.